pontoon/base/views.py

```diff
@@ -660,6 +667,12 @@ def update_translation(request):
     user = request.user
     project = e.resource.project
 
+    def set_user_custom_homepage():
+        """Save custom homepage for user if it's not set"""
+        if not user.profile.custom_homepage:
+            user.profile.custom_homepage = l
+            user.profile.save()
+            
```

Comment:
Saving translations needs to be blazing fast, so we should get rid of all non-essential operations.

It should be enough that users can set their custom homepage in settings. DB migration is a plus.